architecture/additional_concepts/nuagesdn.adoc

```diff
@@ -0,0 +1,117 @@
+
+[[integration-components]]
+
+== Integration Components
+
+Nuage VSP integrates with OpenShift using two main components:
+
+. nuage-openshift-monitor
+. vsp-openshift plugin
+
+[[nuage-openshift-monitor]]
+
+=== nuage-openshift-monitor
+
+The nuage-openshift-monitor is a service that monitors the {product-title} API server for creation of projects, services, users, user-groups etc.
+
+[NOTE]
+=====
+In case of Highly Available (HA) {product-title} cluster with multiple masters, nuage-openshift-monitor process runs on all the masters independently without any change in functionality.
+=====
+
+For the developer workflow, the nuage-openshift-monitor also auto-creates VSD objects by exercising the VSD REST API to map OpenShift constructs to VSP constructs. Each cluster instance maps to a single domain in Nuage VSP. This allows a given enterprise to potentially have multiple cluster installations - one per domain instance for that Enterprise in Nuage. Each Openshift project is mapped to a zone in the domain of the cluster on the Nuage VSP. Whenever the nuage-openshift-monitor sees an addition/deletion of the project, it instantiates a zone using the VSDK APIs corresponding to that project and allocates a block of subnet for that zone. Additionally, the nuage-openshift-monitor also creates a network macro group for this project. Likewise, whenever the nuage-openshift-monitor sees an addition/deletion of a service, it creates a network macro corresponding to the service IP and assigns that network macro to the network macro group for that project (user provided network macro group using labels is also supported) to enable communication to that service.
+
+For the developer workflow, all pods that get created within the zone get IPs from that subnet pool. The Subnet pool allocation and management is done by the nuage-openshift-monitor based on a couple of plugin specific parameters in the master-config file. However the actual IP address resolution and vport policy resolution is still done by VSD based on the domain/zone that gets instantiated when the project is created. If the initial subnet pool is exhausted, the nuage-openshift-monitor carves out an additional subnet from the cluster CIDR to assign to a given project.
+
+For the operations workflow, the users specify Nuage recognized labels on their application/pod specification to resolve the pods into specific user-defined zones/subnets. This however cannot be used to resolve pods in the zones/subnets created via the developer workflow by nuage-openshift-monitor.
+
+[NOTE]
+=====
+In the operations workflow, the administrator is responsible for pre-creating the VSD constructs to map the pods into a specific zone/subnet as well as allow communication between OpenShift entities (ACL rules, policy groups, network macros, and network macro groups). Detailed description of how to use Nuage labels is provided in the link:http://support.alcatel-lucent.com[Nuage VSP Openshift Integration Guide].
```

Comment:
abhat is correct. It is important to ensure that Red Hat is NOT distributing any Nuage software. However, since SDN configuration is something that needs to be accounted for in advance of an install for the best customer experience, configuration documentation should be included for this partner. This would be akin to how we describe configuration of 3rd party storage plugins, for example. The instructions for installing Nuage itself should remain with Nuage.